app/src/lib/stores/app-store.ts

```diff
@@ -1961,16 +1962,26 @@ export class AppStore {
       }
 
       const state = this.getRepositoryState(repository)
+      const tip = state.branchesState.tip
 
-      if (state.branchesState.tip.kind === TipState.Unborn) {
+      if (tip.kind === TipState.Unborn) {
         throw new Error('The current branch is unborn.')
```

Comment:
I think we should have a more user-friendly error message here. Something that will give them the path to success.

Reply:
We really shouldn't let the user get into this state in the first place (this is driven from the push/pull/fetch contextual button).

Reply:
But it's also unrelated to this change. Feel free to open an issue if you want to take it further.